test/libsolidity/smtCheckerTests/bmc_coverage/msg_value_4.sol

```diff
@@ -0,0 +1,17 @@
+pragma experimental SMTChecker;
+contract A {
+    uint x = msg.value;
+    constructor() {
+        assert(x == 0); // should hold
+    }
+}
+
+contract B {
+    constructor() payable {
+        assert(msg.value == 0); // should fail
```

Comment:
I may be missing something here but is it illegal to pass zero value while deploying a contract with a payable constructor?

Reply:
Ah sorry, it fails because `msg.value >= 0` is the correct constraint in this case. Could you please add one such test case for a passing test if one does not exist already?

Reply:
It is part of the `range_check.sol ` test case.